### PR TITLE
Adding ability to GPG sign commits and tags

### PR DESCRIPTION
**Pull request by cdoughty77**
Let me know your thoughts.  Thanks.

### CODE REVIEW

**Comment by cdoughty-r7** [comment]
@rderoldan1  Updated to fix failing tests.  Please let me know if you have any comments or suggestions.  Thanks!

**Comment by cdoughty-r7** [comment]
Updated to fix failing tests. Please let me know if you have any comments or suggestions. Thanks!

File: lib/rubygems/commands/bump_command.rb [hunk 1]
```diff
@@ -81,7 +83,7 @@ def escape(string)
 
     def commit
       say "Creating commit" unless quiet?
-      system("git commit -m \"Bump to #{@new_version_number}\"")
+      options[:sign] ? system("git commit -S -m \"Bump to #{@new_version_number}\"") : system("git commit -m \"Bump to #{@new_version_number}\"")
```
Comment (dgeb) [inline lib/rubygems/commands/bump_command.rb]: A bit verbose for a one-liner. Could you please break this out with an `if` / `else` ?

File: lib/rubygems/commands/tag_command.rb [hunk 2]
```diff
@@ -29,7 +31,7 @@ def execute
 
     def tag
       say "Creating git tag #{tag_name}" unless quiet?
-      system("git tag -am \"tag #{tag_name}\" #{tag_name}")
+      options[:sign] ?  system("git tag -m \"tag #{tag_name}\" -s #{tag_name}") : system("git tag -am \"tag #{tag_name}\" #{tag_name}")
```
Comment (dgeb) [inline lib/rubygems/commands/tag_command.rb]: Same as above.

**Comment by dgeb** [comment]
@cdoughty77 Seems useful - thanks for the PR. Could you please add some tests that use the `sign` option?

**Comment by cdoughty-r7** [comment]
Hah, if you look at the history it was originally in an if/else but @rderoldan1 suggested slimming it to a one liner.  I'll add some tests.  If we really care about the one liner it can be reverted, just let me know.

**Comment by dgeb** [comment]
@cdoughty77 ugh ... sorry to ask you to redo work, but I think there's a limit to how much we should fit in a one liner or ternary expression. I do favor the `if / else` implementation combined with `return` statements. As I see it, the problem with the original `if / else` implementation is that it's not clear that values are being returned.

Also, thanks in advance for the tests and squashing.

**Comment by svenfuchs** [comment]
Thanks for the work here, @cdoughty-r7 @cdoughty77, and sorry for letting this slip.

I have ported your work to `2.0.0.pre.2` (https://github.com/svenfuchs/gem-release/pull/64) in https://github.com/svenfuchs/gem-release/pull/64/commits/d620c2634680ae3c8daf161d29c4c3dd4a80e7aa. I have also used the git `author` information from your last commit in order to give credit on our git history.

If you had a minute to see if this looks like it might be doing what you intended then that would be awesome.

Please let me know!

Closing this one here for now.

**Comment by cdoughty-r7** [comment]
@svenfuchs  I'm not at present able to test this, however the code looks like the equivalent features are present.  Thanks for adding!

**Pull request closed without merging**